src/efi.rs

```diff
@@ -321,10 +321,16 @@ impl Component for Efi {
         log::debug!("Found metadata {}", meta.version);
         let srcdir_name = component_updatedirname(self);
         let ft = crate::filetree::FileTree::new_from_dir(&src_root.sub_dir(&srcdir_name)?)?;
-        // get esp device via /dev/disk/by-partlabel
-        let esp_device = self
-            .get_esp_device()
-            .ok_or_else(|| anyhow::anyhow!("Failed to find ESP device"))?;
+
+        // Get esp device via /dev/disk/by-partlabel
+        // If failed, change to sfdisk
```

Comment:
I would add a lot more to this comment. Basically all of what is in the commit message.

```
During `bootc install to-disk` the esp device is not mounted but
we can call `get_esp_device()` and it will find the device via
the `/dev/disk/by-partlabel/` symlink. When running `cosa build` and
building a disk image using `OSBuild` the EFI partitions is mounted
but no udev `/dev/disk/by-partlabel/` symlinks exist because it's 
just a chroot and not a full linux environment. In that case
fallback to using `blockdev` to find the partition.
```